aiaccel/scheduler/pylocal_scheduler.py

```diff
@@ -58,12 +62,21 @@ def inner_loop_main_process(self) -> bool:
 
         for trial_id in trial_ids:
             self._serialize(trial_id)
-            if self.num_node > 1:
-                th = threading.Thread(target=self.execute, args=(trial_id,))
-                th.start()
-            else:
+            if self.num_node <= 1:
                 self.execute(trial_id)
 
+        if self.num_node > 1:
+            args = []
+            for trial_id in trial_ids:
+                self.storage.trial.set_any_trial_state(trial_id=trial_id, state='running')
+                xs = self.run.get_any_trial_xs(trial_id)
+                args.append([trial_id, xs])
+            results = self.pool.map(self.execute_wrapper, args)
+            for trial_id, (xs, y, err, start_time, end_time) in zip(trial_ids, results):
```

Comment:
```suggestion
            for trial_id, (xs, y, err, start_time, end_time) in zip(
                trial_ids,
                self.pool.map(self.execute_wrapper, args)
            ):
```

また、mapだと同期実行されてしまうので、imap_unordered使うように修正お願いします。

Reply:
レビューありがとうございます。
imap_unordered を使う形で改修しました。
https://github.com/aistairc/aiaccel/blob/92fb2e1187d05ee9b5efb5f9e87a6d1a3ae0ed5b/aiaccel/scheduler/pylocal_scheduler.py#L56-L59